feat: started adding rabbitmq functionality

devanbenz opened the pull request:
- Started adding Queue configuration class 
- Adds docker compose update for rabbitmq

lazyguru commented:
I added a comment, but deferring to @jgrim for full review

jgrim commented on docs/docker-compose.yml:
```diff
@@ -51,6 +51,12 @@ services:
       SUBLINKS_JWT_SECRET: <JWT-SECRET>
       SUBLINKS_BASE_URL: https://demo.sublinks.org
       SUBLINKS_PICTRS_URL: http://pictrs:8080
+      FEDERATION_QUEUE_ENDPOINT: localhost
```
I prefer the nomenclature FEDERATION_QUEUE_HOST rather than FEDERATION_QUEUE_ENDPOINT.

Pdzly reviewed (dismissed):
I cant see anything bad. As you already fixed what jgrim marked.

devanbenz commented:
Added a few changes 
1. Added Json message deserialization to Rabbit listener factory 
2. Modified it so the Producer and Consumer are using different queues (wouldn't make sense for them both to use the same queue 😆 )

lazyguru commented:
> I added a comment, but deferring to @jgrim for full review

I just now realized my comment was "pending" as I hadn't clicked "start review" instead of "add comment" 🤦🏻. You resolved it already though as it was about you only having one queue and needing to add the second queue for bi-directional communication

lazyguru commented on docs/docker-compose.yml:
```diff
@@ -51,6 +51,13 @@ services:
+      FEDERATION_QUEUE_PASS: sublinks
+      FEDERATION_QUEUE_NAME: federation # TODO: Need to get correct queue name
+      BACKEND_QUEUE_NAME: backend # TODO: Need to get correct queue name
+      BACKEND_TOPIC_NAME: backend # TODO: Need to get correct topic name
```
I don't think these need to be TODO items. Using variables here makes it configurable and having the default be backend/federation is probably fine

devanbenz replied:
Sounds good will remove comments

devanbenz commented:
> > I added a comment, but deferring to @jgrim for full review
> 
> 
> 
> I just now realized my comment was "pending" as I hadn't clicked "start review" instead of "add comment" 🤦🏻. You resolved it already though as it was about you only having one queue and needing to add the second queue for bi-directional communication

Haha sounds good!
